### ISSUE TITLE
URI.withinString matches `www.`

### ISSUE THREAD

**Issue by Bonuspunkt** [medialize/URI.js#327]
code
``` js
const URI = require("urijs");
const text = "world wide web is abbreviated with www.";
URI.withinString(text, function(url) {
	console.log(text);
	console.log(url);
});
```
output
```
world wide web is abbreviated with www.
www
```

**Comment by rodneyrehm** [comment]
Hehe, I just fixed that in my local rewrite-URI-in-ES6-playground the other day :D - for what it's worth, `http://.` is another false positive…

**Comment by rodneyrehm** [comment]
@Bonuspunkt I looked at the referenced issue's code and saw your `const incorrectDetections = ["www"];` hack. You could've used `URI.withinString(text, callback, { ignore: /^www\.$/ });`.

Could you explain the `commonSchemes` thing you've got going there? what exactly are you doing here?

**Comment by Bonuspunkt** [comment]
came from an issue with `look at...http://example.com` (correctly) returning `at...http://example.com`.
so if a scheme ends with ex. `http` everything before that gets ignored when returning the url (would be in the example `http://example.com`)

**Comment by rodneyrehm** [comment]
In an attempt to fix bad *user input* `at..http://example.org` (missing space after punctuation), your current implementation accidentally breaks valid URLs like `svn+ssh://example.org`. Although I'm not aware of many protocols in the form of `svn+ssh`, and I've not witnessed this in the wild, most systems allow applications to register custom schemes. Even browsers allow web apps to handle custom schemes like [`web+gustav`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler). Bad input is bad input and assuming users would quickly realize their mistake, I'd not try to fix this.

I'll have a fix for the original `www.` problem shortly

**Comment by rodneyrehm** [comment]
fix released as [v1.18.7](https://github.com/medialize/URI.js/releases/tag/v1.18.7) :)

**Comment by Bonuspunkt** [comment]
❤️ 

thanks for pointing out the `svn+ssh`.
the thing is, its used in an irc client, send messages can't be edited an you are >90% on the receiving end.
so this ends up as UX vs. correctness.
